samples/csharp/chatapp/Startup.cs

```diff
@@ -24,13 +25,16 @@ public Startup(IConfiguration configuration)
         }
 
         public IConfiguration Configuration { get; }
-        
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddAzureClients(builder =>
-            {
-                builder.AddWebPubSubServiceClient(Configuration["Azure:WebPubSub:ConnectionString"], "chat");
-            });
+            services.AddWebPubSub(o => o.ValidationOptions.Add(Configuration["Azure:WebPubSub:ConnectionString"]))
+                .AddAzureClients(builder =>
+                {
+                    builder.AddWebPubSubServiceClient(Configuration["Azure:WebPubSub:ConnectionString"], nameof(SampleChatHub));
+                });
         }
```

Comment:
Why does this API need the connection string twice?

Reply:
`AddWebPubSub` is for the service -> server direction traffic, and the connection string is for [`Abuse Protection`](https://github.com/cloudevents/spec/blob/v1.0.1/http-webhook.md#4-abuse-protection) checks(based on host) and connection Signature checks(based on `AccessKey` and `ConnectionId`) use.

The `AzureClient` is for the server -> service direction traffic. They are now separate and don't strict bind together. Customer can use different services for in/out, though this may not be a common use.

Reply:
This doesn't look very friendly to use even if there's a logical explanation. Can we make this work without 2 explicit calls that each need the same connection string?

Reply:
The connection string for middleware is optional, if customer doesn't have security concerns and good to skip the checks, they can simply add as `services.AddWebPubSub()`.

To share same connection string in the middleware is a little complicated. We're not able to get the `AccessKey` from DI directly. So the other way is to include rest client into the middleware and make them using same connection string. But this will introduce complex for the middleware, for example, all the rest client settings need to be added and whether we need to support in/out on different services. 

For this initial version, shall we make it simple and provide customer flexibility to combine the in/out?